dlib/dnn/layers_abstract.h

```diff
@@ -4089,7 +4089,152 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
-}
+    class positional_encodings_
```

Comment:
Ha yeah wait a minute.  This class is already in this file.  So this duplicates it.  You added it in https://github.com/davisking/dlib/pull/3021.

It looks like all the changes in this PR right now are this extra copy and some formatting and moving stuff around.  I.e. no actual changes.  Maybe we abandon this PR since you already landed what you want in that other PR?   I feel like there was some change in here that was new that you wanted to land in the beginning, but now I forget what it was and it's been lost?